Submitted By: Jim Gifford (patches at jg555 dot com)
Date: 2005-11-20
Initial Package Version: 0.43
Origin: Gentoo
Upstream Status: Unknown
Description: Adds support for R14000
             Hardcodes console to 9600 bps

 
diff -Naur arcload-0.43.orig/loader/detect.c arcload-0.43/loader/detect.c
--- arcload-0.43.orig/loader/detect.c	2005-09-05 17:32:44.000000000 +0000
+++ arcload-0.43/loader/detect.c	2005-11-21 05:39:04.000000000 +0000
@@ -30,6 +30,7 @@
 	{0x0004, "r4000"},
 	{0x0009, "r10000"},
 	{0x000e, "r12000"},
+	{0x000f, "r14000"},
 	{0x0020, "r4600"},
 	{0x0021, "r4700"},
 	{0x0023, "r5000"},
@@ -194,7 +195,7 @@
 	if(var && var[0]=='d') {
 		var=ArcGetEnvironmentVariable("dbaud");
 		if(!var)
-			return NULL;
+			return "baud=9600";
 		sprintf(baudrate,"baud=%s",var);
 		return baudrate;
 	}
